sklearn/neighbors/tests/test_lmnn.py

```diff
@@ -0,0 +1,677 @@
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, max_impostors=1,
+                                      impostor_store='sparse')
+    lmnn.fit(iris_data, iris_target)
+
+
+def test_neighbors_params():
+    from scipy.spatial.distance import hamming
+
+    params = {'algorithm': 'brute', 'metric': hamming}
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, neighbors_params=params)
+    lmnn.fit(iris_data, iris_target)
+    components_hamming = lmnn.components_
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3)
+    lmnn.fit(iris_data, iris_target)
+    components_euclidean = lmnn.components_
+
+    assert not np.allclose(components_hamming, components_euclidean)
+
+
+def test_impostor_store():
+    k = 3
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=k,
+                                      init='identity', impostor_store='list')
+    lmnn.fit(iris_data, iris_target)
+    components_list = lmnn.components_
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=k,
+                                      init='identity', impostor_store='sparse')
+    lmnn.fit(iris_data, iris_target)
+    components_sparse = lmnn.components_
+
+    assert_array_almost_equal(components_list, components_sparse,
+                              err_msg='Toggling `impostor_store` results in '
+                                      'a different solution.')
+
+
+def test_callback():
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, callback='my_cb')
+    assert_raise_message(ValueError,
+                         '`callback` is not callable.',
+                         lmnn.fit, iris_data, iris_target)
+
+    max_iter = 10
+
+    def my_cb(transformation, n_iter):
+        rem_iter = max_iter - n_iter
+        print('{} iterations remaining...'.format(rem_iter))
+
+    # assert that my_cb is called
+    old_stdout = sys.stdout
+    sys.stdout = StringIO()
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, callback=my_cb,
+                                      max_iter=max_iter, verbose=1)
+    try:
+        lmnn.fit(iris_data, iris_target)
+    finally:
+        out = sys.stdout.getvalue()
+        sys.stdout.close()
+        sys.stdout = old_stdout
+
+    # check output
+    assert '{} iterations remaining...'.format(max_iter-1) in out
+
+
+def test_store_opt_result():
+    X = iris_data
+    y = iris_target
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, max_iter=5,
+                                      store_opt_result=True)
+    lmnn.fit(X_train, y_train)
+    transformation = lmnn.opt_result_.x
+    assert_equal(transformation.size, X.shape[1]**2)
+
+
+def test_verbose():
+    # assert there is proper output when verbose = 1
+    old_stdout = sys.stdout
+    sys.stdout = StringIO()
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, verbose=1)
+    try:
+        lmnn.fit(iris_data, iris_target)
+    finally:
+        out = sys.stdout.getvalue()
```

Comment:
you could use `capsys` instead (see for instance):

https://github.com/scikit-learn/scikit-learn/blob/d31b67f23d3d785b17261ada293710380683bc82/sklearn/neighbors/tests/test_nca.py#L369

Reply:
Done.